pkgs/build-support/fetchnodemodules/default.nix

```diff
@@ -0,0 +1,102 @@
+{ lib, stdenvNoCC, makeWrapper, nodejs }:
+
+{ src
+, hash ? ""
+, sha256 ? ""
+, name ? null
+, production ? true
+, runScripts ? false
+, makeTarball ? true
+, preferLocalBuild ? true
+, exposeBin ? false
+, npmFlags ? ""
+, ...
+} @ args:
+
+assert makeTarball -> !exposeBin;
+let
+  suffix = if makeTarball then ".tar.gz" else "";
+  name_ = if name == null then "${src.name}-node_modules${suffix}" else name;
+  nodejs_ = if args ? nodejs then args.nodejs else nodejs;
+
+  hash_ =
+    if hash != "" then { outputHashAlgo = null; outputHash = hash; }
+    else if sha256 != "" then { outputHashAlgo = "sha256"; outputHash = sha256; }
+    else throw "fetchNodeModules requires a hash for ${name_}";
+
+  customArgs = [ "name" "hash" "sha256" ];
+in stdenvNoCC.mkDerivation ({
+  inherit src production runScripts makeTarball preferLocalBuild;
```

Comment:
you forgot to inherit exposeBin, or am I missing something ?

Reply:
saeme with the npmFlags ?

Reply:
Last comment on this, I don't know what is the best practice here. But rather than inheriting variables that are unrelated to mkDerivation (I had to search the codebase to see if those variables where used elsewhere), could it make sense to use those variables directly where they are required? Something like
- remove the inherit runScripts
- inside the build phase have something like
```
''
if  ! ${runScripts} ; then
        NPM_FLAGS+=" --ignore-scripts"
fi
''
```

Reply:
Yes, I should add it there for clarity, even though it's pulled in with the other arguments in the last line with `//`.